### PR TITLE
feat: Add RemainingDecontaminationTime property

### PR DESCRIPTION
**Pull request by hayabusa255**
## Description
**Describe the changes** 
Added a new static property RemainingDecontaminationTime to calculate the remaining time for the decontamination process.

**What is the current behavior?** (You can also link to an open issue here)
Currently, there is no easy way to get the remaining time for the decontamination process.

**What is the new behavior?** (if this is a feature change)
The new RemainingDecontaminationTime property provides an easy way to get the remaining time for the decontamination process.

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No, this PR does not introduce any breaking changes. It adds a new property without modifying existing functionality.

**Other information**:

<br />

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [x] I have checked the project can be compiled
- [x] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [ ] I have checked no IL patching errors in the console

### Other
- [ ] Still requires more testing

### CODE REVIEW

**Review by louis1706 — COMMENTED** [review]
Is this the real time or the time in game ? 

(15sec or 12.5 sec at the start of decontamination)

**Comment by hayabusa255** [comment]
This property provides the actual remaining decontamination time from the start to the end of the game.

**Comment by louis1706** [comment]
> This property provides the actual remaining decontamination time from the start to the end of the game.

still my question the real time on world time

or the time in game (that go a lot faster than IRL

**Comment by hayabusa255** [comment]
The in-game time progresses faster than real-world time. If the actual decontamination time in the game is set for 15 minutes, it will be displayed as 11 minutes in-game to reflect this faster progression accurately.

This property convert to decontamination time to actual real time

**Comment by Cocoa2219** [comment]
If the description of this property isn’t clear, feel free to change it.

File: EXILED/Exiled.API/Features/Map.cs [hunk 1]
```diff
@@ -74,6 +74,50 @@ public static class Map
...
+                        {
+                            totalTime += (float)(controller.DecontaminationPhases[i].TimeTrigger -
+                                                 controller.DecontaminationPhases[i - 1].TimeTrigger);
+                        }
+                    }
+                }
+
+                return decontaminationStarted ? totalTime : 0f;
+            }
+        }
```
Comment (louis1706) [inline EXILED/Exiled.API/Features/Map.cs:119]:
```suggestion
        public static float RemainingDecontaminationTime => DecontaminationController.Singleton.DecontaminationPhases[DecontaminationController.Singleton.DecontaminationPhases.Length - 1].TimeTrigger - DecontaminationController.GetServerTime;
```

**Review by louis1706 — COMMENTED** [review]
doesn't this exact same value ?

File: EXILED/Exiled.API/Features/Map.cs (hunk 1 above)
Comment (hayabusa255) [reply EXILED/Exiled.API/Features/Map.cs:119]: It's genius code!
But I'm not sure if this code actually works as expected.
The decontamination controller can be null or overridden.
When decontamination ends, this code could return negative values.
Still, this code is very clever.
We need to test this to verify it works as expected.

Comment (louis1706) [reply EXILED/Exiled.API/Features/Map.cs:119]: you can still add the check you need

**Comment by VALERA771** [comment]
Pls implement yamato's suggestion, edit it if you want

**Review by VALERA771 — APPROVED** [review]

**Pull request merged**